### ISSUE TITLE
janggi compile ERROR

### ISSUE THREAD

**Issue by d4soft2021** [ianfab/Fairy-Stockfish#274]
I uploarded issue #272, and thanks for prompt answer.
I compiled with (#define LARGEBOARDS  // for janggi large board).
I got success to get exe file, but lot of warning message as follows.
  -----  warning C4293: '<<': shift may be negative or too big
  it caused from  bitboard.h  in
    #ifdef LARGEBOARDS
        constexpr Bitboard Rank9BB = Rank1BB << (FILE_NB * 8);
       constexpr Bitboard Rank10BB = Rank1BB << (FILE_NB * 9);
    #endif

I tried to compile win32 (not 64 bit), therefore shift bit 64 times error (32 bit integer cannot be shifted 64 times)
How can I fix those problems ?

I tried to compile on Visual Studio 2017 Windows.
Thank you for help.

**Comment by ianfab** [comment]
I do not have experience with compilation using Visual Studio, and the recommended setup for Windows is MSYS2/mingw, see https://github.com/ianfab/Fairy-Stockfish/wiki/Compiling-Fairy-Stockfish#compiling-on-windows. However, in principle compilation with Visual Studio should work, since the continuous integration also tests that, see https://github.com/ianfab/Fairy-Stockfish/blob/d2449ba4624dc20a0570e5ea39ce6f15a0d2ac2e/appveyor.yml#L34-L66.

Does the compiled executable work, or does it crash when you are using it?

**Comment by d4soft2021** [comment]
I compiled Visual Studio 17 windows, but I don't use MSYS2/mingw.
I added all cpp and h files to project (exclude python) and compiled them (I don't use makefile styles)
set linker stack size as you described.
32 bit (x86) setting.

Without Largeboard Everthing OK except minor several warning messages,
But with #define largeboards many warning messages come up.

I don't understand the role of Bitboard, therefore bit shift of Bitboard ERROR is not up to me.
My plan is 
   after compiling, follow step by step debugging.
   I cannot go further without understanding why Bitboard shift is needed.
   If I understand Bitboard role, I can ignore warning messages.
I think to try x64 compiling too, because Bitboard shift is OK for long integer.
However winboard says engine must be compiled 32 bit  method.

Thank you.
